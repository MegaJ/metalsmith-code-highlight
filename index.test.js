/* eslint-env jest,node */
const metalsmithCodeHighlight = require('./index');
const {promisify} = require('util');

const files = {
  'escape.html': {
    contents: new Buffer('<code class=lang-js>true && false</code>'),
  },
  'code.html': {
    contents: new Buffer('<code class=lang-js>// Hi</code>'),
  },
  'double.html': {
    contents: new Buffer(
      `
          <p>Hello there.</p>
          <p>Inline <code class="lang-js other">document.all</code></p>
          <pre class="lang-highlight"><code class=lang-coffeescript>
          \nrequire "fs"\nconsole.log fs.readFileSync "/etc/passwd"
          </code></pre>
        `,
    ),
  },
  'doctype.html': {
    contents: new Buffer(
      `
          <!DOCTYPE html>
          <html>
          <head><title>Test Page</title></head>
          <body>
          <pre class="lang-highlight"><code class=lang-js>var x = [1, 2, 3];</code></pre>
          <code class=lang-bogus>What language is this?</code>
          </body>
          </html>
        `,
    ),
  },
};

beforeEach(() => {
  return promisify(metalsmithCodeHighlight({useFragments: true}))(files, {});
});

for (let file in files) {
  test(file, () => {
    expect(files[file].contents.toString()).toMatchSnapshot();
  });
}
