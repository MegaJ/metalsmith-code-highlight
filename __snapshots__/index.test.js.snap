// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code.html 1`] = `"<code class=\\"lang-js hljs javascript\\"><span class=\\"hljs-comment\\"><span class=\\"hljs-comment\\">// Hi</span></span></code>"`;

exports[`doctype.html 1`] = `
"<!DOCTYPE html><html><head><title>Test Page</title></head>
      <body class=\\"lang-highlight\\">
      <pre class=\\"lang-highlight\\"><code class=\\"lang-js hljs javascript\\"><span class=\\"hljs-keyword\\"><span class=\\"hljs-keyword\\"><span class=\\"hljs-keyword\\"><span class=\\"hljs-keyword\\">var</span></span></span></span> x = [<span class=\\"hljs-number\\"><span class=\\"hljs-number\\"><span class=\\"hljs-number\\"><span class=\\"hljs-number\\">1</span></span></span></span>, <span class=\\"hljs-number\\"><span class=\\"hljs-number\\"><span class=\\"hljs-number\\"><span class=\\"hljs-number\\">2</span></span></span></span>, <span class=\\"hljs-number\\"><span class=\\"hljs-number\\"><span class=\\"hljs-number\\"><span class=\\"hljs-number\\">3</span></span></span></span>];</code></pre>
      <code class=\\"lang-bogus\\">What language is this?</code>
      
      
    </body></html>"
`;

exports[`double.html 1`] = `
"
      <p>Hello there.</p>
      <p class=\\"lang-highlight\\">Inline <code class=\\"lang-js other hljs javascript\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\">document</span></span></span>.all</code></p>
      <pre class=\\"lang-highlight\\"><code class=\\"lang-coffeescript hljs\\">
      
<span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\">require</span></span></span> <span class=\\"hljs-string\\"><span class=\\"hljs-string\\"><span class=\\"hljs-string\\">\\"fs\\"</span></span></span>
<span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\">console</span></span></span>.log fs.readFileSync <span class=\\"hljs-string\\"><span class=\\"hljs-string\\"><span class=\\"hljs-string\\">\\"/etc/passwd\\"</span></span></span>
      </code></pre>
    "
`;

exports[`escape.html 1`] = `"<code class=\\"lang-js hljs javascript\\"><span class=\\"hljs-literal\\">true</span> &amp;&amp; <span class=\\"hljs-literal\\">false</span></code>"`;

exports[`readme.html 1`] = `
"
      <p>Hello there.</p>
      <p class=\\"lang-highlight\\">Inline <code class=\\"lang-js hljs javascript\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\">document</span></span></span></span></span>.all</code></p>
      <pre class=\\"lang-highlight\\"><code class=\\"lang-coffeescript hljs\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\">require</span></span></span></span></span> <span class=\\"hljs-string\\"><span class=\\"hljs-string\\"><span class=\\"hljs-string\\"><span class=\\"hljs-string\\"><span class=\\"hljs-string\\">\\"fs\\"</span></span></span></span></span>
      <span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\"><span class=\\"hljs-built_in\\">console</span></span></span></span></span>.log fs.readFileSync <span class=\\"hljs-string\\"><span class=\\"hljs-string\\"><span class=\\"hljs-string\\"><span class=\\"hljs-string\\"><span class=\\"hljs-string\\">\\"/etc/passwd\\"</span></span></span></span></span><span class=\\"hljs-string\\"><span class=\\"hljs-string\\"><span class=\\"hljs-string\\"><span class=\\"hljs-string\\"><span class=\\"hljs-string\\">'</span></span></span></span></span></code></pre>
    "
`;
